src/components/calcite-time-picker/calcite-time-picker.tsx

```diff
@@ -0,0 +1,801 @@
+import {
+  Component,
+  Element,
+  Host,
+  h,
+  Prop,
+  VNode,
+  Event,
+  EventEmitter,
+  Watch,
+  State,
+  Listen,
+  Method
+} from "@stencil/core";
+import { Scale, Theme } from "../interfaces";
+import {
+  Meridiem,
+  zeroPadNumber,
+  MinuteOrSecond,
+  numberKeys,
+  stringIsValidNumber,
+  Time,
+  maxTenthForMinuteAndSecond,
+  TimeComponents
+} from "../../utils/time";
+import { CSS } from "./resources";
+
+@Component({
+  tag: "calcite-time-picker",
+  styleUrl: "calcite-time-picker.scss",
+  shadow: true
+})
+export class CalciteTimePicker {
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+
+  @Element() el: HTMLCalciteTimePickerElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  //--------------------------------------------------------------------------
+
+  /** The hour value (24-hour format) */
+  @Prop({ reflect: true, mutable: true }) hour?: string = null;
+
+  /** Format of the hour value (12-hour or 24-hour) (this will be replaced by locale eventually) */
+  @Prop({ reflect: true }) hourDisplayFormat: "12" | "24" = "12";
+
+  /** aria-label for the hour input */
+  @Prop() intlHour = "hour";
+
+  /** aria-label for the hour down button */
+  @Prop() intlHourDown = "decrease hour";
+
+  /** aria-label for the hour up button */
+  @Prop() intlHourUp = "increase hour";
+
+  /** aria-label for the meridiem (am/pm) input */
+  @Prop() intlMeridiem = "AM/PM";
+
+  /** aria-label for the meridiem (am/pm) down button */
+  @Prop() intlMeridiemDown = "decrease AM/PM";
+
+  /** aria-label for the meridiem (am/pm) up button */
+  @Prop() intlMeridiemUp = "increase AM/PM";
+
+  /** aria-label for the minute input */
+  @Prop() intlMinute = "minute";
+
+  /** aria-label for the minute down button */
+  @Prop() intlMinuteDown = "decrease minute";
+
+  /** aria-label for the minute up button */
+  @Prop() intlMinuteUp = "increase minute";
+
+  /** aria-label for the second input */
+  @Prop() intlSecond = "second";
+
+  /** aria-label for the second down button */
+  @Prop() intlSecondDown = "decrease second";
+
+  /** aria-label for the second up button */
+  @Prop() intlSecondUp = "increase second";
+
+  /** The minute value */
+  @Prop({ reflect: true, mutable: true }) minute?: string = null;
```

Comment:
I'm wondering if the typing for these should be number instead of string? Should it be typed to be two places or less?

Reply:
They have to be string because they need to be stored with leading zeros.  I looked into this and the browser time inputs do the same thing.